### PR TITLE
ns-api: add backup APIs

### PR DESCRIPTION
**Pull request by Tbaile**
Added APIs for:
- set passphrase
- backup
- restore (with or without password)
- remote backup list
- remote backup
- remote restore (with or without password)
- download remote backup

### CODE REVIEW

File: packages/ns-api/files/ns.backup [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+if cmd == 'list':
+    print(json.dumps({
+        'backup': {
+            'passphrase': 'str'
+        },
+        'restore': {
+            'backup': 'str',
+            'passphrase': 'str'
+        },
+        'registered-set-passphrase': {
```
[the remark this answers is not in the document]
Comment (Tbaile) [reply packages/ns-api/files/ns.backup:91]: I've used the `subscription` prefix cause passphrase is fetched only if you have a registered subscription, can change, no issues

File: packages/ns-api/files/ns.backup [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            # restore backup
+            restore_backup(data.get('passphrase', None))
+            # restore successful, restart the system
+            subprocess.run(['/sbin/reboot'], check=True, capture_output=True)
+            # reboot takes a few seconds to complete, enough to send the response
+            print(json.dumps({'message': 'success'}))
+        except RuntimeError as error:
+            print(json.dumps(utils.generic_error(error.args[0])))
+
+    elif action == 'registered-set-passphrase':
```
[the remark this answers is not in the document]
Comment (Tbaile) [reply packages/ns-api/files/ns.backup:132]: K, will do

File: packages/ns-api/files/ns.backup [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        except subprocess.CalledProcessError as error:
+            print(json.dumps(utils.generic_error(f'remote list failed')))
+
+    elif action == 'registered-backup':
+        try:
+            # create backup with passphrase set before if present
+            passphrase = fetch_enterprise_passphrase()
+            create_backup(passphrase=passphrase)
+            # upload backup to server and remove it from filesystem
+            completed_process = subprocess.run(['/usr/sbin/remote-backup', 'upload', BACKUP_PATH], check=True,
```
[the remark this answers is not in the document]
Comment (Tbaile) [reply packages/ns-api/files/ns.backup:155]: I've started this feature calling `send-backup`, however I've found few issues with it:
- is used by a cron, if changes are needed to this executable will be reflected to a one-time button in the UI, not so good.
- if there's a backup with same SHA, upload won't be executed at all
- returns 0 if there's no subscription present, not the best error code to find out if something went wrong

Besides, this approach is more connected to everything else, both the UI and the CLI run the same executable

File: packages/ns-api/files/ns.backup [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        except IOError as error:
+            print(json.dumps(utils.generic_error(f'passphrase write failed')))
+        except KeyError as error:
+            print(json.dumps(utils.validation_error('passphrase', 'required')))
+
+    elif action == 'registered-list-backups':
+        try:
+            # list remote backups, the return is a string, so it gets decoded and encoded to generate a formatted json
+            completed_process = subprocess.run(['/usr/sbin/remote-backup', 'list'], check=True, capture_output=True)
+            print(json.dumps({'values': json.loads(completed_process.stdout.decode('utf-8'))}))
```
Comment (gsanchietti) [inline packages/ns-api/files/ns.backup:149]: Just a matter of taste, but you could avoid calling `decode`:
```suggestion
            completed_process = subprocess.run(['/usr/sbin/remote-backup', 'list'], check=True, capture_output=True, text=True)
            print(json.dumps({'values': json.loads(completed_process.stdout)}))
```

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**